Makefile

```diff
@@ -116,13 +127,64 @@ verify-codegen: download-deps
 	./hack/verify-codegen.sh
 
 # Build the docker image
-docker-build:
+docker-build: vendor
 	docker build -f Dockerfile.dev . -t ${IMG}
 
 # Push the docker image
 docker-push:
 	docker push ${IMG}
 
+.PHONY: e2e-test-dependencies
+e2e-test-dependencies:
+	./scripts/download-e2e-test-dependencies.sh
+
+.PHONY: e2e-test-env
+e2e-test-env: export PATH:=./bin:${PATH}
+e2e-test-env: e2e-test-dependencies
+	kind create cluster
+
+	# TODO get these from the MetalLB resource yaml
+	docker pull metallb/controller:v0.9.6
+	docker pull metallb/speaker:v0.9.6
+	kind load docker-image metallb/controller:v0.9.6
+	kind load docker-image metallb/speaker:v0.9.6
+	./scripts/install-metallb.sh
+
+	docker pull kennethreitz/httpbin
+	kind load docker-image kennethreitz/httpbin
+
+	# TODO collect these from the operator (run the related functions and collect the referenced images)
+	docker pull gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+	docker pull docker.io/istio/pilot:${ISTIO_VERSION}
+	docker pull docker.io/istio/proxyv2:${ISTIO_VERSION}
```

Comment:
We probably should pull from ghcr now.

Reply:
So, these must be the same as the images in `config/samples/istio_v1beta1_istio.yaml`. I think this would be an unrelated change and should be done in a separate PR.

Reply:
Okay, sure. It's fine for me btw until we don't hit issues with Docker rate limiting, but we should migrate at some point.